infra/modules/gcp-host/main.tf

```diff
@@ -160,7 +143,7 @@ module "api_connector" {
 
   secret_bindings = merge(
     # bc some of these are later filled directly, bind to 'latest'
-    { for k, v in module.secrets[each.key].secret_bindings : k => merge(v, { version_number : "latest" }) },
+   {for k, secrets in local.secrets_to_provision[each.key] : k => merge({ secret_id : module.secrets[each.key].secret_ids_within_project[k], version_number : "latest" }) if (try(!secrets.lockable, true) && try(!secrets.writable, true))},
```

Comment:
yeah, I agree with this - good change, if not very readable to do it this way 😓  - it's correct via the interface we've defined.